### PR TITLE
Use local population for microbe spawns

### PR DESCRIPTION
**Pull request by Oliveriver**
**Brief Description of What This PR Does**

Use local rather than global population to determine microbe species spawn densities.

A bug that somehow slipped under everyone's noses for several years. This explains why the player species always seemed to have such a high spawn rate. This change will also mean the player species (other than the player cell and its daughter) won't spawn if the player moves to a patch they haven't yet colonised.

@buckly90 This may need a balancing pass. I picked 25 as the divisor, but there's not a lot of reasoning behind it. It needs to be less than 50 (otherwise the player species won't spawn for the next generation as well) and I went a fair bit lower to cancel out the reduction in spawn rate caused by using local rather than global populations.

**Related Issues**

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/microbe_stage/PatchManager.cs [hunk 1]
```diff
@@ -181,6 +181,7 @@ private void HandleCellSpawns(Patch patch)
         foreach (var entry in patch.SpeciesInPatch.OrderByDescending(entry => entry.Value))
         {
             var species = entry.Key;
+            var population = entry.Value;
 
             if (species.Population <= 0)
```
Comment (hhyyrylainen) [inline src/microbe_stage/PatchManager.cs:186]: Shouldn't this be updated as well to use `population`?

**Review by RacerBG — APPROVED** [review]
Looks good.

1. When I move to a different patch, indeed my cell and my daughter cell are the only inhabitants of the new patch.
2. My initial population is always 50 cells, after that it easily grows beyond 2000 for 1 generation.

**Comment by Oliveriver** [comment]
> Looks good.
> 
> 1. When I move to a different patch, indeed my cell and my daughter cell are the only inhabitants of the new patch.
> 2. My initial population is always 50 cells, after that it easily grows beyond 2000 for 1 generation.

What do you think of the spawn rate of cells in general?

**Comment by RacerBG** [comment]
> > Looks good.
> > 
> > 1. When I move to a different patch, indeed my cell and my daughter cell are the only inhabitants of the new patch.
> > 2. My initial population is always 50 cells, after that it easily grows beyond 2000 for 1 generation.
> 
> What do you think of the spawn rate of cells in general?

It should be lower, it's getting high too fast. I'm not talking about specifically about this PR though this is my general opinion for the spawn rate.

**Comment by buckly90** [comment]
I think it works perfectly fine.

Player species is no longer absolutely everywhere, and I had no trouble finding colony candidates either. Splendid job!

**Review by Kasterisk — APPROVED** [review]

**Pull request merged**